blog/_posts/2018-04-12-in-a-nutshell.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: blog-page
+title: Scala in a (tasty) Nutshell
+author: Martin Odersky and Nicolas Stucki
+authorImg: /images/martin.jpg
+date: 2018-03-05
+---
+
+One of the biggest open questions for migrating to Scala 3 is what to do about macros. In this blog post we give our current thinking, which is to try to achieve full alignment between macros and Tasty.
+
+## What is TASTY?
+
+Tasty is the high-level interchange format for Scala 3. It is based on <u>t</u>yped <u>a</u>bstract <u>s</u>yntax <u>t</u>rees. These trees contain in a sense all the information present in a Scala program. They represent the syntactic structure of programs in every detail and also contain the complete information about types and positions. The Tasty "snapshot" of a code file is taken after type checking (so that all types are present and all implicits are elaborated) but before any transformations (so that no information is lost or changed). The file representation of these trees is heavily optimized for compactness, which means that we can generate full Tasty trees on every compiler run and rely on nothing else for supporting separate compilation.
+
+The information present in TASTY trees can be used for many purposes.
+
+ - The compiler uses it to support separate compilation.
+ - A language server for an IDE uses it to support hyperlinking, command completion, or documentation.
```

Comment:
has documentation support been implemented in tasty? Comment below says it works today.

Reply:
I think we are working on it